src/wast-parser.y

```diff
@@ -665,13 +666,41 @@ expr1 :
       assert(if_->type == ExprType::If);
       if_->if_.true_->label = $2;
     }
-  | try_check labeling_opt LPAR BLOCK block RPAR catch_list {
-      Expr* try_ = Expr::CreateTry($5, $7.first);
-      try_->try_block.label = $2;
-      $$ = join_exprs1(&@1, try_);
+  | try_check labeling_opt try_ {
+      Block* block = $3->try_block.block;
+      block->label = $2;
+      $$ = join_exprs1(&@1, $3);
     }
   ;
 
+try_ :
+    block_sig try_ {
+      $$ = $2;
+      Block* block = $$->try_block.block;
+      $$->try_block.block->sig.insert(block->sig.end(), $1->begin(), $1->end());
+      delete $1;
+    }
+  | try_inst_list
+  ;
+
+try_inst_list :
+      instr catch_list {
```

Comment:
No, I suppose not. Though that seems to be an inconsistency with the flat format. But maybe we can just add a TODO and figure it out later.